python/extensions/private/requirements.bzl.tmpl

```diff
@@ -29,5 +33,14 @@ def dist_info_requirement(name):
 def entry_point(pkg, script = None):
     """entry_point returns the target of the canonical label of the package entrypoints.
     """
-    # TODO: https://github.com/bazelbuild/rules_python/issues/1262
-    print("not implemented")
+    selects = _entry_point(
+        tmpl = "%%ENTRYPOINT_TMPL%%",
+        pkg = _clean_name(pkg),
+        script = script,
+        packages = _packages,
+        default_version = _default_py_version,
+    )
+    if selects == None:
+        fail("Package '{}' does not exist, select one from: {}".format(pkg, _packages.keys()))
+
+    return select(selects)
```

Comment:
I am not sure it is possible without fetching all of the repos for a particular package for all python versions. This solution allows the user to only fetch what is truly needed.

I would expect the user to normally create an alias themselves based on the returned value from this function. We could change it to generate an alias and return nothing instead now that I am thinking about it because that is the only useful way to use the `entry_point` macro.

Another option would be the user passing a list of `pkg, script` pairs to the `pip.parse` extension so that we can generate the aliases for those in the `pip_hub` repository. That way using the `entry_point` macro would be optional.

Reply:
Ah, right. Ok, using a select() expression is pretty clever to avoid that :).

And I see that needing use_repo() is avoided because the RHS of the select expressions are canonical repos. Also clever!

> We could change it to generate an alias and return nothing instead now that I am thinking about it because that is the only useful way to use the entry_point macro.

Is it, though? I'm not entirely sure. True, the pip docs do suggest putting it behind an alias, but isn't that just for convenience? It isn't strictly required to do so.

Another option would be to generate an alias, but also return the label to that alias. This is fairly close to ideal. Though, the downside is we have to auto-generate a name to preserve the "entry_point() can be called multiple times" behavior (this should be easy with native.existing_rules()).

Argh. I think, for now, let's just go with your initial approach here to return a select expression.

> Another option would be the user passing a list of pkg, script pairs to the pip.parse

Eh, yeah, I suppose. I'm not necessarily against this; it's just that there's a couple FR's for various ways to modify the BUILD files for a pip package, and we already have a somewhat large and complex API to do so already -- more ideas in this space make me think we need to revisit the feature more holistically.

Reply:
I'll add a comment about this in the `entry_point.bzl` docs.

Reply:
I suggested a brief comment. We can put more discussion/rationale in the issue.